### PR TITLE
Make resolving some exception types optional in `CFGTranslationPhaseOne`

### PR DESCRIPTION
**Pull request by cushon**
Fixes https://github.com/typetools/checker-framework/issues/4452

### CODE REVIEW

**Review by mernst — CHANGES_REQUESTED** [review]
Could you please write some documentation (probably on `maybeGetTypeMirror` to explain the reason for the method and the reason that the method might return null?  Otherwise someone might be confused by the code, and might simplify the code by removing the method.

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java [hunk 1]
```diff
@@ -3754,4 +3764,20 @@ public Node visitOther(Tree tree, Void p) {
...
+
+    /**
+     * Returns the TypeMirror for the given class, or {@code null}.
+     *
+     * @param clazz a class
+     * @return the TypeMirror for the class
+     */
+    private @Nullable TypeMirror maybeGetTypeMirror(Class<?> clazz) {
+        String name = clazz.getCanonicalName();
+        assert name != null : "@AssumeAssertion(nullness): assumption";
```
Comment (mernst) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java:3776]: The method only works for classes with a canonical name.  That is worth mentioning in its Javadoc, and then here "assumption" can be more specific.

Comment (cushon) [reply dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java]: Done

**Review by cushon — COMMENTED** [review]
> Could you please write some documentation (probably on `maybeGetTypeMirror` to explain the reason for the method and the reason that the method might return null?

Done

**Review by mernst — APPROVED** [review]

**Comment by mernst** [comment]
Thanks!

**Pull request merged**